static/src/js/components/SpatialDisplay/SpatialDisplay.js

```diff
@@ -309,481 +230,476 @@ class SpatialDisplay extends Component {
+    const pointArray = currentPolygonSearch.length ? currentPolygonSearch[0].split(',') : []
+    const pointCount = (pointArray.length / 2) - 1
+
+    entry = (
+      <SpatialDisplayEntry>
+        {
+          pointCount > 2 && (
+            <Row className="spatial-display__form-row">
+              <span
+                className="spatial-display__text-primary"
+                data-testid="spatial-display_polygon"
+              >
+                {`${pointCount} ${pluralize('Point', pointCount)}`}
+              </span>
+            </Row>
+          )
+        }
+      </SpatialDisplayEntry>
+    )
+
+    if (pointArray.length < 2) {
+      hint = 'Draw a polygon on the map to filter results'
     }
 
-    if (!items.length) {
-      return null
+    secondaryTitle = 'Polygon'
+
+    if (displaySpatialPolygonWarning) {
+      spatialError = 'This collection does not support polygon search. Your polygon has been converted to a bounding box.'
     }
 
-    return (
-      // eslint-disable-next-line react/jsx-no-useless-fragment
-      <>
-        {items}
-      </>
-    )
+    contents.push((
+      <FilterStackContents
+        key="filter__polygon"
+        body={entry}
+        title="Polygon"
+      />
+    ))
+  } else if (((currentLineSearch && currentLineSearch.length) && !drawingNewLayer) || drawingNewLayer === 'polyline') {
+    entry = <SpatialDisplayEntry />
+
+    contents.push((
+      <FilterStackContents
+        key="filter__polygon"
+        body={entry}
+        title="Line"
+      />
+    ))
+  }
+
+  if (contents.length) {
+    items.push((
+      <FilterStackItem
+        key="item__spatial"
+        dataTestId="filter-stack__spatial"
+        icon={FaCrop}
+        title="Spatial"
+        secondaryTitle={secondaryTitle}
+        error={drawingNewLayer && !manuallyEnteringVal ? '' : spatialError}
+        onRemove={onSpatialRemove}
+        hint={hint}
+      >
+        {contents}
+      </FilterStackItem>
+    ))
+  }
+
+  if (!items.length) {
+    return null
   }
+
+  return (
+    // eslint-disable-next-line react/jsx-no-useless-fragment
+    <>
+      {items}
+    </>
+  )
 }
 
 SpatialDisplay.propTypes = {
   boundingBoxSearch: PropTypes.arrayOf(PropTypes.string).isRequired,
   circleSearch: PropTypes.arrayOf(PropTypes.string).isRequired,
+  defaultError: PropTypes.string,
   displaySpatialPolygonWarning: PropTypes.bool.isRequired,
   drawingNewLayer: PropTypes.oneOfType([
     PropTypes.string,
     PropTypes.bool
   ]).isRequired,
   lineSearch: PropTypes.arrayOf(PropTypes.string).isRequired,
+  manuallyEntering: PropTypes.string,
   onChangeQuery: PropTypes.func.isRequired,
   onRemoveSpatialFilter: PropTypes.func.isRequired,
   pointSearch: PropTypes.arrayOf(PropTypes.string).isRequired,
   polygonSearch: PropTypes.arrayOf(PropTypes.string).isRequired,
   shapefile: PropTypes.shape({}).isRequired
 }
 
+SpatialDisplay.defaultProps = {
```

Comment:
nitpick; the default props got before the prop declarations in all the other namespaces